libevm/JitVM.cpp

```diff
@@ -173,15 +177,35 @@ int64_t call(
 	params.data = input;
 	params.onOp = {};
 
-	auto output = env.call(params);
-	auto gasLeft = static_cast<int64_t>(params.gas);
-
-	output.second.copyTo({_outputData, _outputSize});
-	if (!output.first)
-		// Add failure indicator.
-		gasLeft |= EVM_CALL_FAILURE;
-
-	return gasLeft;
+	bool success = false;
+	owning_bytes_ref output;
+	std::tie(success, output) = env.call(params);
+	// FIXME: We have a mess here. It is hard to distinguish reverts from failures.
+	// In first case we want to keep the output, in the second one the output
+	// is optional and should not be passed to the contract, but can be useful
+	// for EVM in general.
+	o_result->code = success ? EVM_SUCCESS : EVM_REVERT;
+	o_result->gas_left = static_cast<int64_t>(params.gas);
+
+	// Pass the output to the EVM without a copy. The EVM will delete it
+	// when finished with it.
+
+	// First assign reference. References are not invalidated when vector
+	// of bytes is moved. See `.takeBytes()` below.
+	o_result->output_data = output.data();
+	o_result->output_size = output.size();
+
+	// Place a new vector of bytes containing output in result's reserved memory.
+	static_assert(sizeof(bytes) <= sizeof(o_result->reserved), "Vector is too big");
+	new(&o_result->reserved) bytes(output.takeBytes());
```

Comment:
Yeah, there's one allocation less and one indirection less, but in general such esoteric tricks should better be used only when we're really sure we need them. 

In general placement new should better not be used outside of lower level memory-managing utility classes like containers, I guess.

But hm, ok if you wanted to try it  :) I'm ok with leaving it like this for now